rotkehlchen/chain/solana/manager.py

```diff
@@ -0,0 +1,95 @@
+import logging
+from collections.abc import Callable, Sequence
+from typing import TYPE_CHECKING, TypeVar
+
+from httpx import HTTPError
+from solana.rpc.api import Client
+from solana.rpc.core import RPCException
+from solders.pubkey import Pubkey
+
+from rotkehlchen.chain.constants import DEFAULT_RPC_TIMEOUT
+from rotkehlchen.chain.evm.types import WeightedNode
+from rotkehlchen.chain.mixins.rpc_nodes import SolanaRPCMixin
+from rotkehlchen.chain.solana.utils import lamports_to_sol
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.fval import FVal
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import SolanaAddress, SupportedBlockchain
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.greenlets.manager import GreenletManager
+
+R = TypeVar('R')
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class SolanaManager(SolanaRPCMixin):
+
+    def __init__(
+            self,
+            greenlet_manager: 'GreenletManager',
+            database: 'DBHandler',
+    ):
+        SolanaRPCMixin.__init__(self)
+        self.greenlet_manager = greenlet_manager
+        self.database = database
+        self.blockchain = SupportedBlockchain.SOLANA
+        self.rpc_timeout = DEFAULT_RPC_TIMEOUT
+
+    def _query(
+            self,
+            fn: Callable[[Client], R],
+            call_order: Sequence[WeightedNode],
+    ) -> R:
+        """Use the provided call order to request fn from solana"""
```

Comment:
What is fn? A funciton? Do they call ti like this? Why not be more explicit and write `function`?

Reply:
This is for the thing I did to have the type checking instead of having `getattr(node, 'query_multi_account')`. Will rename to method.